src/OmniSharp.Host/CompositionHostBuilder.cs

```diff
@@ -116,7 +122,11 @@ private static IEnumerable<Type> SafeGetTypes(Assembly a)
             IServiceCollection services = null,
             Action<ILoggingBuilder> configureLogging = null)
         {
-            services = services ?? new ServiceCollection();
+            services ??= new ServiceCollection();
+
+            services.TryAddSingleton(_ => new ManualFileSystemWatcher());
+            services.TryAddSingleton<IFileSystemNotifier>(_ => _.GetRequiredService<ManualFileSystemWatcher>());
+            services.TryAddSingleton<IFileSystemWatcher>(_ => _.GetRequiredService<ManualFileSystemWatcher>());
```

Comment:
this is an odd way to use a discard. why not a regular variable name e.g. `s => s.GetRequiredService`?

Reply:
I dunno, I probably copied it from some old aspnet core code?  I've been doing that for a long while, but I can change it here.